source/math/FloatPoint2D.ooc

```diff
@@ -32,6 +32,10 @@ FloatPoint2D: cover {
 		(this scalarProduct(other) / (this norm * other norm)) clamp(-1, 1) acos() * (this x * other y - this y * other x < 0 ? -1 : 1)
 	}
 	distance: func (other: This) -> Float { (this - other) norm }
+	distanceSquared: func (other: This) -> Float {
+		point := this - other
```

Comment:
Yes, I though about this too, but in the end I chose `point` mostly because it is a `FloatPoint2D` in terms of data representation. But I will have to agree that `distanceVector` maybe is more representative. Will change.